ecs-agent/tmds/handlers/fault/v1/types/types.go

```diff
@@ -13,8 +13,59 @@
 
 package types
 
+import "fmt"
+
 const (
-	BlackHolePortFaultType = "network-blackhole-port"
-	LatencyFaultType       = "network-latency"
-	PacketLossFaultType    = "network-packet-loss"
+	BlackHolePortFaultType    = "network-blackhole-port"
+	LatencyFaultType          = "network-latency"
+	PacketLossFaultType       = "network-packet-loss"
+	missingRequiredFieldError = "required parameter %s is missing"
 )
+
+type NetworkFaultRequest interface {
+	ValidateRequest() error
+	ToString() (string, error)
+}
+
+type NetworkBlackholePortRequest struct {
+	Port        *int    `json:"Port"`
+	Protocol    *string `json:"Protocol"`
+	TrafficType *string `json:"TrafficType"`
+}
+
+type NetworkFaultInjectionResponse struct {
+	Status string `json:"status,omitempty"`
+	Error  string `json:"error,omitempty"`
+}
+
+func (request NetworkBlackholePortRequest) ValidateRequest() error {
+	if request.Port == nil {
+		return fmt.Errorf(missingRequiredFieldError, "Port")
+	}
+	if request.Protocol == nil || *request.Protocol == "" {
+		return fmt.Errorf(missingRequiredFieldError, "Protocol")
+	}
+	if request.TrafficType == nil || *request.TrafficType == "" {
+		return fmt.Errorf(missingRequiredFieldError, "TrafficType")
+	}
+
+	return nil
+}
+
+func (request NetworkBlackholePortRequest) ToString() (string, error) {
+	if err := request.ValidateRequest(); err != nil {
+		return "", err
+	}
```

Comment:
It's odd that `ToString` method validates the request too. Why do we need a `ToString` method? Can we not use Go's built-in struct stringification?

Reply:
I guess the goal was to check for nil pointers. Fine, I can just use built-in struct stringy

Reply:
Also, IIRC it doesn't work quite as nice with pointer fields. But if you think this is better we can go with this

Reply:
Makes sense, addressed in the latest commit